salt/modules/boto_s3_bucket.py

```diff
@@ -0,0 +1,831 @@
+              "Prefix": "prefixstring",\\
+              "Status": "enabled",\\
+              "Transitions": [{...},],\\
+              "NoncurrentVersionTransitions": [{...},],\\
+              "NoncurrentVersionExpiration": {...},\\
+        }]'
+
+    '''
+
+    try:
+        conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+        if Rules is not None and isinstance(Rules, string_types):
+            Rules = json.loads(Rules)
+        conn.put_bucket_lifecycle_configuration(Bucket=Bucket, LifecycleConfiguration={'Rules': Rules})
+        return {'updated': True, 'name': Bucket}
+    except ClientError as e:
+        return {'updated': False, 'error': salt.utils.boto3.get_error(e)}
+
+
+def put_logging(Bucket,
+           TargetBucket=None, TargetPrefix=None, TargetGrants=None,
+           region=None, key=None, keyid=None, profile=None):
+    '''
+    Given a valid config, update the logging parameters for a bucket.
+
+    Returns {updated: true} if parameters were updated and returns
+    {updated: False} if parameters were not updated.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_s3_bucket.put_logging my_bucket log_bucket '[{...}]' prefix
+
+    '''
+
+    try:
+        conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+        logstate = {}
+        for key, val in {
+                'TargetBucket': TargetBucket,
+                'TargetGrants': TargetGrants,
+                'TargetPrefix': TargetPrefix,
+        }.iteritems():
+            if val is not None:
+                logstate[key] = val
+        if logstate:
+            logstatus = {'LoggingEnabled': logstate}
+        else:
+            logstatus = {}
+        if TargetGrants is not None and isinstance(TargetGrants, string_types):
+            TargetGrants = json.loads(TargetGrants)
+        conn.put_bucket_logging(Bucket=Bucket, BucketLoggingStatus=logstatus)
+        return {'updated': True, 'name': Bucket}
+    except ClientError as e:
+        return {'updated': False, 'error': salt.utils.boto3.get_error(e)}
+
+
+def put_notification_configuration(Bucket,
+           TopicConfigurations=None, QueueConfigurations=None,
+           LambdaFunctionConfigurations=None,
+           region=None, key=None, keyid=None, profile=None):
+    '''
+    Given a valid config, update the notification parameters for a bucket.
+
+    Returns {updated: true} if parameters were updated and returns
+    {updated: False} if parameters were not updated.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_s3_bucket.put_notification_configuration my_bucket
+                [{...}] \\
+                [{...}] \\
+                [{...}]
+
+    '''
+
+    try:
+        conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+        if TopicConfigurations is None:
+            TopicConfigurations = []
+        elif isinstance(TopicConfigurations, string_types):
```

Comment:
It is using six.string_types. Do you mean for me to change the import so it's literally six.string_types?